wenet/dataset/dataset.py

```diff
@@ -187,3 +187,31 @@ def Dataset(data_type,
     dataset = Processor(dataset, processor.batch, **batch_conf)
     dataset = Processor(dataset, processor.padding)
     return dataset
+
+
+class IPUCollateFn:
+    """
+    # TODO move those static padding into Dataset
+    """
+
+    def __init__(self, max_feature_length, max_target_length, type):
+        self.max_feature_length = max_feature_length
+        self.max_target_length = max_target_length
+        assert type in ['float32', 'float16']
+        self.type = type
+
+    def __call__(self, batch):
+        batch_size = len(batch)
+        feature_length = torch.cat([i[3] for i in batch])
+        target_length = torch.cat([i[4] for i in batch])
+        padded_feature = torch.zeros(
+            size=[batch_size, self.max_feature_length, batch[0][1].size(2)])
```

Comment:
What's the difference of this code compared to the pading function in processor.py?

Reply:
the padding shape is statics while the original function is dynamic